shallow_backup.py

```diff
@@ -535,7 +535,8 @@ def git_add_all_commit(repo, dir_path):
 	packages_path = os.path.join(dir_path, "packages")
 	configs_path = os.path.join(dir_path, "configs")
 	gitignore_path = os.path.join(dir_path, ".gitignore")
-	repo.index.add([gitignore_path])
+	if os.path.exists(gitignore_path):
```

Comment:
It's not possible for this path to not exist, so this check is unnecessary.

The `gitignore` is created at the start of each run if it doesn't exist. The way to stop this is by adding a blank `gitignore`.